src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_COUNT = 5;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(FIGURE_COUNT);
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case 0:
+                double firstLeg = random.nextDouble() * 10 + 1;
+                double secondLeg = random.nextDouble() * 10 + 1;
+                double trapezoidHeight = random.nextDouble() * 10 + 1;
```

Comment:
Creating variables for the random values right before passing them to the constructor is good for readability, but consider encapsulating the logic for generating these values within private helper methods to further improve code organization.